src/app/scenario/scenario.component.ts

```diff
@@ -431,12 +521,64 @@ export class ScenarioComponent implements OnInit {
 
   doDeleteVMSet() {
     this.selectedscenario.virtualmachines.splice(this.deletingVMSetIndex, 1);
+    if (Object.keys(this.selectedscenario.virtualmachines[0]).length == 0)
```

Comment:
Your suggestion seems good, unfortunately, when I implemented it, in case I delete the virtual machine, and I have none, I should not be able to go to the next step, but I am able to go to the next step, which we want to avoid at all cost.

Reply:
See my newest comment, this will resolve the issue + remove the variable :)

Reply:
Also, is always this.selectedscenario.virtualmachines.length == 1, because by default, we add this.selectedscenario.virtualmachines = {}, so the solution that I have implemented, will see what is the element inside {} and count the length of it.

Reply:
Sure @abrahimi93  this works. But what happens if the user deletes this VMSet :) - so better create a function to check if any VM is given

Reply:
Done.